src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -401,7 +401,7 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
      * Returns the latest finished result for modeling and text exercises. It does not have to be rated.
      * For other exercise types it returns a rated result.
      */
-    getLatestRatedResult() {
+    getLatestRatedResult(): void {
```

Comment:
the methods name does not make sense. getSomething should not have a void return type.
Either return something or call it e.g. `loadLatestRatedResult`.

However it's also questionable that `this.complaintService.findBySubmissionId` is invoked here when "just the latest rated result should be retrieved. So in fact, this method rather loads the complaint based on the submission id for all exercises with an existing result and potentially updates the latest result for quiz or programming exercises.

When you touch the method, I think you should refactor it, because currently it is really bad code (not your fault) and the JSDoc is also horrible :-(

Reply:
I renamed the function and the javadoc to better reflect the function's behavior.